### DISCUSSION TITLE
Out of date authentication API version for Kubernetes?

### DISCUSSION THREAD

**Discussion by yorinasub17** [gruntwork-io/knowledge-base#478] [Help] [answered]
I am getting the following error message when using the Gruntwork modules:

```
Kubernetes cluster unreachable: exec plugin: invalid apiVersion "client.authentication.k8s.io/v1alpha1"
```

How do I address this?

---

<ins datetime="2022-06-17T23:42:22Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108805">Tracked in ticket #108805</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
This is due to the Kubernetes library dependencies of the `helm` provider being upgraded beyond 1.24.x, where the `v1alpha1` API was removed.

We are currently working on a fix for `kubergrunt` to be compatible with generating a `v1beta1` auth token. In the meantime, it is recommended to work around this by pinning the `helm` provider version to `2.5.1`, the last known version to work with `v1alpha1`. You can pin the provider version by following the same strategy recommended in https://github.com/gruntwork-io/knowledge-base/discussions/187.

Refer to https://github.com/hashicorp/terraform-provider-helm/issues/893 for more info.

**Comment by jalessio** [comment]
@yorinasub17 Thanks for the info!

If anyone is looking for a quick fix, this worked for us in the top-level `terragrunt.hcl` file:

```
generate "helm_provider_version" {
  path      = "helm_version_override.tf"
  if_exists = "overwrite_terragrunt"
  contents  = <<EOF
terraform {
  required_providers {
    helm = {
      source  = "hashicorp/helm"
      version = "= 2.5.1"
    }
  }
}
EOF
}
```
